src/submission/submission-to-clinical/stat-calculator.ts

```diff
@@ -85,14 +87,19 @@ const calcDonorCoreEntityStats = (
 
     if (tumorAndNormalExists) {
       coreStats[schemaNameToCoreCompletenessStat[clinicalType]] =
-        donor.specimens.map(sp => sp.clinicalInfo).filter(notEmpty).length / donor.specimens.length;
+        donor.specimens
+          .filter(
+            specimen =>
+              specimen && specimen.sampleType && dnaSampleTypes.includes(specimen.sampleType),
+          )
+          .filter(notEmpty).length / donor.specimens.length;
```

Comment:
Can we add a comment here explaining why we are applying this filter?

Should we consider making a named filter function (like with `notEmpty`) that can be reused?
```ts
const dnaSpecimenFilter = (specimen?: Specimen): boolean => specimen && specimen.sampleType && dnaSampleTypes.includes(specimen.sampleType);
```

Reply:
Comments are always important, added

Reply:
Reusable function added